modules/videoio/test/test_video_io.cpp

```diff
@@ -309,30 +343,53 @@ Ext_Fourcc_PSNR synthetic_params[] = {
 #ifdef HAVE_MSMF
 
 #if !defined(_M_ARM)
-    makeParam("wmv", "WMV1", 30.f),
-    makeParam("wmv", "WMV2", 30.f),
+    makeParam("wmv", "WMV1", 30.f, CAP_MSMF),
+    makeParam("wmv", "WMV2", 30.f, CAP_MSMF),
 #endif
-    makeParam("wmv", "WMV3", 30.f),
-    makeParam("avi", "H264", 30.f),
-    makeParam("wmv", "WVC1", 30.f),
+    makeParam("wmv", "WMV3", 30.f, CAP_MSMF),
+    makeParam("avi", "H264", 30.f, CAP_MSMF),
+    makeParam("wmv", "WVC1", 30.f, CAP_MSMF),
 
 #else // HAVE_MSMF
```

Comment:
`#else` is not needed here, because it will disable all other tests. Please use the scheme as follows:
```.cpp
#ifdef HAVE_MSMF
    makeParam("wmv", "WMV1", 30.f, CAP_MSMF);
    //... all lines with CAP_MSMF
#endif

#ifdef HAVE_FFMPEG
    // all lines with CAP_FFMPEG
#endif

...
```